bindings/sftp/sftp.go

```diff
@@ -0,0 +1,314 @@
+package sftp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+
+	sftpClient "github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/knownhosts"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/components-contrib/metadata"
+	"github.com/dapr/kit/logger"
+	kitmd "github.com/dapr/kit/metadata"
+)
+
+const (
+	metadataRootPath = "rootPath"
+	metadataFileName = "fileName"
+)
+
+// Sftp is a binding for file operations on sftp server.
+type Sftp struct {
+	metadata   *sftpMetadata
+	logger     logger.Logger
+	sftpClient *sftpClient.Client
+}
+
+// sftpMetadata defines the sftp metadata.
+type sftpMetadata struct {
+	RootPath       string `json:"rootPath"`
+	FileName       string `json:"fileName"`
+	Address        string `json:"address"`
+	Username       string `json:"username"`
+	Password       string `json:"password"`
+	PrivateKey     []byte `json:"privateKey"`
+	HostPublicKey  []byte `json:"hostPublicKey"`
+	KnownHostsFile string `json:"knownHostsFile"`
+	InsecureSSL    bool   `json:"isInsecureSSL"`
```

Comment:
This is actually not SSL-related. More appropriate would be `insecureIgnoreHostKey`